tidb/src/tidb/sql.clj

```diff
@@ -93,6 +93,17 @@
   (close! conn)
   (open (::node conn) (::test conn)))
 
+(defn set-auto-commit!
+  "Set a JDBC connection's autocommit variable, will retry until success."
+  [conn auto-commit]
+  (when-let [c (j/db-find-connection conn)]
+    (loop []
+      (try
+        (.setAutoCommit c auto-commit)
+        (catch Exception e
+          (info "Failed to set auto-commit, error" e "retrying...")
+          (Thread/sleep 100))))))
```

Comment:
Here `loop` seems to be useless since there is no `recur`. I think the following impl is just ok, if `.setAutoCommit` throws an exception, jepsen eventually generates an `:info` event, which then leads to reconnection for the next op.
```clj
(when-let [c (j/db-find-connection conn)]
  (.setAutoCommit c auto-commit))
```